main/src/ca/uwaterloo/flix/language/phase/njvm/Mnemonics.scala

```diff
@@ -89,4 +150,118 @@ object Mnemonics {
 
   }
 
+  //Capabilities
+  /**
+    * Capability which allows to load/store a local variable
+    */
+  sealed trait Local[T]{
+
+    val localType : JvmType
+    val location : Int
+
+    def LOAD[R <: Stack]() : F[R] => F[R**T] =  t => t.load(localType, location)
+    def STORE[R <: Stack]() : F[R**T] => F[R] =  t => t.load(localType, location)
+  }
+
+  class LocalC[T](val localType: JvmType, val location: Int) extends Local[T]
+
+  /**
+    * Capability which allows to get/put a field
+    */
+  sealed trait Field[T]{
+    val fieldName : String
+    val fieldType : JvmType
+
+    def GET_FIELD[R <: Stack]() : F[R ** JvmType.Reference] => F[R ** T] =  t => t.getField(fieldName, fieldType)
+    def PUT_FIELD[R <: Stack]() : F[R ** JvmType.Reference ** T] => F[R] =  t => t.putField(fieldName, fieldType)
+  }
+
+  class FieldC[T](val fieldName: String, val fieldType: JvmType) extends Field[T]
+
+  sealed trait FunSig[R] {
+    val modifiers : List[JvmModifier]
+    val methodName : String
+    val returnType : JvmType
+
+    def getArgsType : List[JvmType]
+  }
+
+  class FunSig0[R](val modifiers : List[JvmModifier], val methodName : String, val returnType : JvmType,
```

Comment:
OK, I like this pattern. But I wonder if the return type cannot be part of the type parameters?

Reply:
Maybe the modifiers should be a set?